### PR TITLE
Use chained exceptions instead of combining them

### PR DESCRIPTION
**Pull request by CaioCoutinhoP**
### Description

I chose some instances to make the change. After running the grep -r 'except .*Error as' astropy/io/ and grep -r 'except Exception as' astropy/io , I discovered these specific instances that need to be modified.

### CODE REVIEW

File: astropy/io/fits/fitsrec.py [hunk 1]
```diff
@@ -879,8 +879,8 @@ def _convert_ascii(self, column, field):
             indx = self.names.index(column.name)
             raise ValueError(
                 "{}; the header may be missing the necessary TNULL{} "
-                "keyword or the table contains invalid data".format(exc, indx + 1)
-            )
+                "keyword or the table contains invalid data".format(indx + 1)
+            )from exc
```
Comment (pllim) [inline astropy/io/fits/fitsrec.py:883]:
```suggestion
            ) from exc
```

**Comment by pllim** [comment]
Thanks!

pre-commit.ci autofix

**Comment by taldcroft** [comment]
@CaioCoutinhoP - I am OK with changing to chained exceptions, with the important point that you **do not change the exception message at all**. For instance in the ECSV error message it is critical to see the reason for the problem in the final exception message. I made a malformed ECSV file and tried to read, and it gives this helpful message:
```
File ~/git/astropy/astropy/io/ascii/ecsv.py:373, in EcsvOutputter._convert_vals(self, cols)
    368     raise ValueError(
    369         f"column {col.name!r} failed to convert: "
    370         "column value is not valid JSON"
    371     )
    372 except Exception as exc:
--> 373     raise ValueError(f"column {col.name!r} failed to convert: {exc}")

ValueError: column 'b' failed to convert: could not convert string to float: 'hello'
```

With your changes, the `could not convert string to float: 'hello'` would be buried several pages up in the traceback, making it difficult for users to see what went wrong.

**Review by taldcroft — CHANGES_REQUESTED** [review]
@CaioCoutinhoP - there are still a few cases where the exception message is being changed. If you revert those then this PR can likely get merged.

**Pull request closed without merging**